spot-contracts/contracts/RouterV2.sol

```diff
@@ -59,77 +59,37 @@ contract RouterV1 {
-        (int256 reserveFee, uint256 protocolFee) = perp.feeStrategy().computeMintFees(mintAmt);
-        int256 mintFee = reserveFee + protocolFee.toInt256();
-        return (mintAmt, feeToken, mintFee);
-    }
-
     /// @notice Tranches the collateral using the current deposit bond and then deposits individual tranches
     ///         to mint perp tokens. It transfers the perp tokens back to the
     ///         transaction sender along with any unused tranches and fees.
     /// @param perp Address of the perp contract.
     /// @param bond Address of the deposit bond.
     /// @param collateralAmount The amount of collateral the user wants to tranche.
-    /// @param feePaid The fee paid to the perp contract to mint perp when the fee token is not the perp token itself, otherwise 0.
-    /// @dev Fee to be paid should be pre-computed off-chain using the preview function.
     function trancheAndDeposit(
         IPerpetualTranche perp,
         IBondController bond,
-        uint256 collateralAmount,
-        uint256 feePaid
+        uint256 collateralAmount
     ) external afterPerpStateUpdate(perp) {
         BondTranches memory bt = bond.getTranches();
         IERC20Upgradeable collateralToken = IERC20Upgradeable(bond.collateralToken());
-        IERC20Upgradeable feeToken = perp.feeToken();
 
         // transfers collateral & fees to router
         collateralToken.safeTransferFrom(msg.sender, address(this), collateralAmount);
-        if (feePaid > 0) {
-            feeToken.safeTransferFrom(msg.sender, address(this), feePaid);
-        }
 
         // approves collateral to be tranched
         _checkAndApproveMax(collateralToken, address(bond), collateralAmount);
 
         // tranches collateral
         bond.deposit(collateralAmount);
 
-        // approves fee to be spent to mint perp tokens
-        _checkAndApproveMax(feeToken, address(perp), feePaid);
+        // uses senior tranches to mint perps
+        uint256 trancheAmt = bt.tranches[0].balanceOf(address(this));
+        _checkAndApproveMax(bt.tranches[0], address(perp), trancheAmt);
+        perp.deposit(bt.tranches[0], trancheAmt);
 
-        for (uint8 i = 0; i < bt.tranches.length; i++) {
-            uint256 trancheAmt = bt.tranches[i].balanceOf(address(this));
-            // approves tranches to be spent
-            _checkAndApproveMax(bt.tranches[i], address(perp), trancheAmt);
-            // mints perp tokens using tranches
-            uint256 mintAmt = perp.deposit(bt.tranches[i], trancheAmt);
-            if (mintAmt <= 0) {
-                // transfers unused tranches back
-                bt.tranches[i].safeTransfer(msg.sender, trancheAmt);
-            }
+        // transfers remaining tranches back
+        for (uint8 i = 1; i < bt.tranches.length; i++) {
+            bt.tranches[i].safeTransfer(msg.sender, bt.tranches[i].balanceOf(address(this)));
         }
 
         // transfers any remaining collateral tokens back
```

Comment:
I think we can safely remove transferring back remaining collateral here

Reply:
Will make a note, will remove as part of the gas efficiency pass later